test/cufisher_price.cu

```diff
@@ -2,12 +2,132 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include <curand.h>
+#include <curand_kernel.h>
+#include <iostream>
+
+#include <AdePT/BlockData.h>
+
+struct MyTrack {
+  int index{0};
+  int pdg{0};
+  double energy{0};
+  double pos[3]{0};
+  double dir[3]{0};
+  bool flag1;
+  bool flag2;
+};
+
+struct Scoring {
+  adept::Atomic_t<int> secondaries;
+  adept::Atomic_t<float> totalEnergyLoss;
+
+  VECCORE_ATT_HOST_DEVICE
+  Scoring() {}
+
+  VECCORE_ATT_HOST_DEVICE
+  static Scoring *MakeInstanceAt(void *addr)
+  {
+    Scoring *obj = new (addr) Scoring();
+    return obj;
+  }
+};
+
+// kernel function that does energy loss or pair production
+__global__ void process(adept::BlockData<MyTrack> *block, Scoring *scor, curandState_t *states)
+{
+  int particle_index = blockIdx.x * blockDim.x + threadIdx.x;
+
+  // check if you are not outside the used block
+  if (particle_index > block->GetNused() + block->GetNholes()) return;
+
+  // check if the particle is still alive (E>0)
+  if ((*block)[particle_index].energy == 0) return;
+
+  // generate random number
+  float r = curand_uniform(states);
+
+  // call the 'process'
+  if (r < 0.5f) {
+    // energy loss
+    float eloss = 0.2f * (*block)[particle_index].energy;
+    scor->totalEnergyLoss.fetch_add(eloss < 0.001f ? (*block)[particle_index].energy : eloss);
+    (*block)[particle_index].energy = (eloss < 0.001f ? 0.0f : ((*block)[particle_index].energy - eloss));
+
+    // if particle dies (E=0) release the slot
+    if ((*block)[particle_index].energy == 0) block->ReleaseElement(particle_index);
+  } else {
+    // pair production
+    float eloss = 0.5f * (*block)[particle_index].energy;
+    (*block)[particle_index].energy -= eloss;
+
+    // here I need to create a new particle
+    auto secondary_track = block->NextElement();
+    if (!secondary_track) return;
```

Comment:
```suggestion
    assert (secondary_track != nullptr && "No slot available for secondary track" );
```
This is a bad situation and the program should crash, otherwise the energy loss will not add up to the primary energy